cmd/frontend/graphqlbackend/survey_response.go

```diff
@@ -119,3 +121,67 @@ func (r *schemaResolver) SubmitSurvey(ctx context.Context, args *struct {
 
 	return &EmptyResponse{}, nil
 }
+
+// FeedbackSubmissionInput contains a happiness feedback response.
+type HappinessFeedbackSubmissionInput struct {
+	// Score is the user's happiness rating, from 1-4.
+	Score int32
+	// Feedback is the answer to "What's going well? What could be better?".
+	Feedback *string
+	// The URL that the happiness feedback was submitted from
+	CurrentURL string
+}
+
+type happinessFeedbackSubmissionForHubSpot struct {
+	Email      *string `url:"email"`
+	Score      int32   `url:"happiness_score"`
+	Feedback   *string `url:"happiness_feedback"`
+	CurrentURL string  `url:"happiness_current_url"`
+	SiteID     string  `url:"site_id"`
+}
+
+// SubmitHappinessFeedback records a new happiness feedback response by the current user.
+func (r *schemaResolver) SubmitHappinessFeedback(ctx context.Context, args *struct {
+	Input *HappinessFeedbackSubmissionInput
+}) (*EmptyResponse, error) {
+	var email *string
+
+	if args.Input.Score < 1 || args.Input.Score > 4 {
+		log15.Error("Score must be a value between 1 and 4")
```

Comment:
I think this should be returned to the caller 
```suggestion
		return nil, errors.New("Score must be a value between 1 and 4")
```